framework/py/flwr/common/inflatable_utils.py

```diff
@@ -0,0 +1,94 @@
+# Copyright 2025 Flower Labs GmbH. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""InflatableObject utils."""
+
+
+from typing import Union
+
+from flwr.proto.fleet_pb2_grpc import FleetStub  # pylint: disable=E0611
+from flwr.proto.message_pb2 import (  # pylint: disable=E0611
+    PullObjectRequest,
+    PullObjectResponse,
+    PushObjectRequest,
+    PushObjectResponse,
+)
+from flwr.proto.serverappio_pb2_grpc import ServerAppIoStub  # pylint: disable=E0611
+
+from .inflatable import InflatableObject, get_object_head_values_from_object_content
+from .record import Array, ArrayRecord, ConfigRecord, MetricRecord
+
+# Helper registry that maps names of classes to their type
+inflatable_class_registry: dict[str, type[InflatableObject]] = {
+    Array.__qualname__: Array,
+    ArrayRecord.__qualname__: ArrayRecord,
+    ConfigRecord.__qualname__: ConfigRecord,
+    MetricRecord.__qualname__: MetricRecord,
+}
+
+
+def push_object_to_servicer(
+    obj: InflatableObject, stub: Union[FleetStub, ServerAppIoStub]
+) -> list[str]:
+    """Recursively deflate an object and push it to the servicer.
+
+    It returns the list of pushed object IDs.
+    """
+    pushed_object_ids = []
+    # Push children if it has any
+    if obj.children:
+        for child in obj.children.values():
+            pushed_object_ids.extend(push_object_to_servicer(child, stub))
+
+    # Deflate object and push
+    object_content = obj.deflate()
+    _: PushObjectResponse = stub.PushObject(
+        PushObjectRequest(
+            object_id=obj.object_id,
+            object_content=object_content,
+        )
+    )
+    pushed_object_ids.append(obj.object_id)
```

Comment:
I think this can be more efficient.

```suggestion
    object_content = obj.deflate()
    object_id = get_object_id(object_content)
    _: PushObjectResponse = stub.PushObject(
        PushObjectRequest(
            object_id=object_id,
            object_content=object_content,
        )
    )
    pushed_object_ids.append(object_id)
```